### ISSUE TITLE
Task: Implement additional Jetpack tests.

### ISSUE THREAD

**Issue by worldomonation** [Automattic/wp-calypso#80730]
### Details

Taking into account outcome for pdWQjU-rL-p2, begin implementing new Jetpack E2E tests from the highest priority & easiest to implement.

### Checklist
Task list taken from the Jetpack - Calypso test coverage mapping spreadsheet.

_No response_
```[tasklist]
### Must Cover Tasks
- [x] Jetpack Stats - navigate to /stats, navigate. https://github.com/Automattic/wp-calypso/pull/80891
- [x] Jetpack Social - configure sharing buttons in Tools > Marketing > Sharing Buttons https://github.com/Automattic/wp-calypso/pull/81073
- [x] Jetpack Social - sharing buttons on published content https://github.com/Automattic/wp-calypso/pull/80897
- [x] Blocks/Design - Layout (It appears it was already covered in blocks__core.ts)
- [x] Jetpack/Social - toggle on/off Blaze. https://github.com/Automattic/wp-calypso/pull/81252
- [x] Jetpack/Traffic - SEO preview https://github.com/Automattic/wp-calypso/pull/81199
- [x] Blocks/Earn - Ad https://github.com/Automattic/wp-calypso/pull/81127
- [x] Blocks/Earn - Payment Button Blocked by https://github.com/Automattic/jetpack/issues/30785
- [x] Blocks/Earn - Donation Form -- this seems to be done!
- [x] Blocks/Misc - AI Assistant https://github.com/Automattic/wp-calypso/pull/80899 https://github.com/Automattic/wp-calypso/pull/81065
- [x] Blocks/Misc - Writing Prompt https://github.com/Automattic/wp-calypso/pull/80957
- [x] Blocks/Media - VIdeoPress https://github.com/Automattic/wp-calypso/pull/81026
- [x] Blocks/Form - Jetpack Button https://github.com/Automattic/wp-calypso/pull/81044
- [x] Blocks/Form - Name https://github.com/Automattic/wp-calypso/pull/81044
- [x] Blocks/Form - Email https://github.com/Automattic/wp-calypso/pull/81044
- [x] Blocks/Form - Multi-line https://github.com/Automattic/wp-calypso/pull/81044
- [x] Form block submission  https://github.com/Automattic/wp-calypso/pull/81301
- [x] Form block AI configured https://github.com/Automattic/wp-calypso/pull/81298
- [x] Jetpack Search https://github.com/Automattic/wp-calypso/pull/80991
- [x] Blocks/Earn - Paywall https://github.com/Automattic/wp-calypso/pull/81256
- [x] Estimated to be difficult below --------------------------------------------------------
- [x] Jetpack Newsletter - Subscription Management https://github.com/Automattic/wp-calypso/pull/81437
- [ ] Jetpack Newsletter - Newsletter settings @worldomonation
- [ ] Jetpack Newsletter - editor sidebar access settings
- [x] Blocks/Earn - Payments https://github.com/Automattic/wp-calypso/pull/81024
- [x] Blocks/Earn - Donations https://github.com/Automattic/wp-calypso/pull/81010
- [x] Jetpack - smoke test pages in the dashboard https://github.com/Automattic/wp-calypso/pull/81443
```

```[tasklist]
### Important Tasks
- [x] Blocks/Embed - Related Posts
- [x] Blocks/Misc - GIF
- [x] Blocks/Misc - Image Compare
- [x] Blocks/Misc - Map
- [x] Blocks/Misc - Story
- [x] Blocks/Misc - Markdown
- [x] Estimated to be difficult below --------------------------------------------------------
- [x] Jetpack Social - Set up connections
- [ ] Jetpack Social - Auto-share to social on publish
- [x] Jetpack Social - Social preview
- [x] Can we check for the Stats pixel.wp.com request on the front-end of WPCOM sites somehow? https://github.com/Automattic/wp-calypso/pull/82913
```

**Comment by dpasque** [comment]
I'll take point on Forms. Asked a question below about designing a potential dedicated E2E spec (as opposed to block smoke test) that focuses on Forms. Let me know what you think!

pdWQjU-rL-p2#comment-537

**Comment by dpasque** [comment]
After some more designing today, I had the idea to split up up the block smoke testing for forms and the full flow.

This means, two specs:
- `blocks__jetpack-forms`, which will itself include three flows...
- - One where you add a pre-canned Contact Form
- - One where you add all the possible fields
- - One where you transform a Form block using the custom pattern options.
- `jetpack__full-form-flow`, which will use a very basic form (like RSVP) and go through submission and validation.

**Comment by worldomonation** [comment]
⬆️  Sounds good to me. 

One nitpick suggestion: what if the full spec is named like `jetpack__rsvp-flow`?

**Comment by dpasque** [comment]
> One nitpick suggestion: what if the full spec is named like jetpack__rsvp-flow?

Oh that's fine too! Although, I'm not sure I want to necessarily couple it tightly to a specific type of form, because we might change that based on whatever is easiest --- the focus isn't the form content, but that you can submit forms on webpages and see the responses in Admin!

**Comment by dpasque** [comment]
Adding some test plan mapping notes / ideas:

> Jetpack Social - Social preview

This test actually shouldn't be that hard, and would be a good test! We could just...
1. Make a new post
2. Open the Jetpack Sidebar
3. Add an "SEO Title"
4. Launch the preview and validate one or two social media previews.

> Jetpack Newsletter - editor sidebar access settings

I added this because I think this is a big money maker for us, and I think any coverage here could be good.
We could set up a paid newsletter plan on these accounts, but we don't necessarily need paid subscribers. Even a test that sets it to paid subscribers and makes sure that any ol' person on the internet can't view the post. We can maybe also have a dedicated user that has a free subscription on all of these sites.


> Jetpack Social - Auto-share to social on publish

It looks like the simple sites may have a tumblr connection already set up, so we might be able to use that to test a very simple/basic version of this case!

**Comment by worldomonation** [comment]
> Can we check for the Stats pixel.wp.com request on the front-end of WPCOM sites somehow?

It looks like we have a tracking pixel and its details are here: p7pQDF-5HJ-p2.

Basically, we want to see if the request for a tracking pixel (at `pixel.wp.com`) succeeded, and that should be enough to verify that at least the frontend loaded the tracking pixel. From our last Jetpack <> KitKat meeting, I'm pretty sure it's going to be difficult to then query whether the tracked stat made its way back to the database.

@jeherve does the above sound okay?

**Comment by jeherve** [comment]
Yes, I think checking for the `https://pixel.wp.com/g.gif` should be enough. You would want to check for specific parameters with that request, since other a8c tracking solutions such as Tracks can also use `g.gif` but with different parameters.

Checking for the `blog` and `post` parameters should be good to ensure this is a Stats tracking pixel.

**Comment by worldomonation** [comment]
> Yes, I think checking for the `https://pixel.wp.com/g.gif` should be enough. You would want to check for specific parameters with that request, since other a8c tracking solutions such as Tracks can also use `g.gif` but with different parameters.
> 
> Checking for the `blog` and `post` parameters should be good to ensure this is a Stats tracking pixel.

Edit: I figured it out for AT sites - the tracking pixels don't load if viewing the post as the publishing user, unlike on Simple where it does. Solution is to view the post as a logged out user.

Oddly, it seems on the AT blogs the tracking pixel isn't being loaded. These are the only requests I see:
- https://pixel.wp.com/g.gif?v=wpcom-no-pv&x_notes-empty-message=all_shown&baba=0.24164228172509428
- https://pixel.wp.com/g.gif?x_stats-initial-visibility=visible&v=wpcom-no-pv&rand=0.6719089094645461
- https://pixel.wp.com/g.gif?v=wpcom-no-pv&x_notes-menu-impressions=zero-jetpack&baba=0.5357180544050155
- https://pixel.wp.com/g.gif?x_stats-initial-visibility=visible&v=wpcom-no-pv&rand=0.8408110455779745

@jeherve

**Comment by jeherve** [comment]
> it seems on the AT blogs the tracking pixel isn't being loaded.

Is the Stats module enabled on those sites?

> Solution is to view the post as a logged out user.

Another option could be to change the Stats options to allow tracking logged in users as well, but that may be more steps.

**Comment by worldomonation** [comment]
> > it seems on the AT blogs the tracking pixel isn't being loaded.
> 
> Is the Stats module enabled on those sites?

Yes, it is enabled. The issue as I found out is a behavior difference between AT and Simple sites.
- Simple: loads tracking pixels even if viewing as the publishing user.
- AT: does not load tracking pixel if viewing as publishing user.

> > Solution is to view the post as a logged out user.
> 
> Another option could be to change the Stats options to allow tracking logged in users as well, but that may be more steps.

To clarify, the tracking pixel is being loaded as logged out user on AT and Simple. The only time the tracking pixel isn't loading on AT is when viewing the published post _as the publishing user_. 

In any case, we want to view the post not as the publishing user.

**Comment by dpasque** [comment]
Going to go ahead and close this now, as we are at the end of our window for implementing these test. We've gotten the vast vast vast majority of the items here, and increased the coverage a lot!
